### ISSUE TITLE
City-States don't reward people who protect them from attacks

### ISSUE THREAD

**Issue by RecursiveVision** [LoneGazebo/Community-Patch-DLL#7932]
Quoting from a forum post:

> Not sure this is intended (or otherwise impossible to program against) but I've had a recent war with the Zulu, who were at the time attacking a CS.
> 
> I entered the CS's territory and destroyed a bunch of the enemy's units, therefore saving their city... but I was getting trespassing penalties instead of being rewarded for attacking a common foe, just as it already happens with major civs.
> 
> This was particulary jarring because at a point the Zulu did manage to capture the CS. I wore down the city's defenses and was about to recapture it... and then the Persians rushed in with cavalry, captured the CS and returned it. So I ended up with a very angry CS that gave the Persians a ridiculous influence bonus (and allyship). I feel like that my previous war effort should have been rewarded instead of losing the diplomacy game to a civ that arrived much later just to collect the spoils.
> 
> TL;DR: "we fought against a common foe" should be implemented for city-states 

City-States should at minimum offer the +Influence that you get for killing nearby Barbarians, and working to liberate a CS only to have someone else zoom in and "ninja liberate" it has bad mouthfeel.

**Comment by Stalker012345** [comment]
> City-States should at minimum offer the +Influence that you get for killing nearby Barbarians, and working to liberate a CS only to have someone else zoom in and "ninja liberate" it has bad mouthfeel.

I think the +influence makes sense....but at the end of the day part of the strategy of the game is timing your attacks to be the one who gets the cake. Its the same with barbs, with CS, with taking enemy cities. When two people fight over the same thing, only one of them gets it, and you have to time yourself to make sure that its you!

So i don't agree that Persians snipe here is a problem....that's just part of the game.

**Comment by RecursiveVision** [comment]
Fair enough, but we could add +Influence and negate the trespassing penalties.

**Comment by Stalker012345** [comment]
So what is the plan on this one? Are changes planned or are we leaving this as not going to change?

**Comment by LoneGazebo** [comment]
sorry, I meant to tag this as fixed. I'm going to include war civs as 'invaders' (a la barbs) so that if they have units in a civ's territory, other civs helping won't incur intrusion penalties.

**Comment by LoneGazebo** [comment]
no bonus influence though, as that's a bear to include. could also potentially be gamed
